packages/core/content-manager/admin/src/preview/components/PreviewContent.tsx

```diff
@@ -0,0 +1,27 @@
+import * as React from 'react';
+
+import { Page } from '@strapi/admin/strapi-admin';
+
+import { usePreviewContext } from '../pages/Preview';
+
+const PreviewContent = () => {
+  const previewUrl = usePreviewContext('PreviewIframe', (state) => state.url);
+
+  if (!previewUrl) {
+    return <Page.Loading />;
+  }
```

Comment:
Does typescript tell you previewUrl is nullable? Because I believe by the time this component renders it should always be available already. If not we need to fix the context's type. But the loading state should be handled at the page level, no need to add it here

Reply:
shouldn't be null! will remove this :)